### PR TITLE
Array Tasks 1.0

### PR DESCRIPTION
**Pull request by Arens404**

### CODE REVIEW

File: Sum.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    }
+    public static int numbersSum(int[] numbersArray){
+        int arrayNumbersSum = 0;
+        for(int i = 0; i < numbersArray.length; i++){
+            arrayNumbersSum += numbersArray[i];
+        }
+        return arrayNumbersSum;
+    }
+    public static void numbersCheck(ArrayList<Integer> numbers) {
```
Comment (Chezo) [inline Sum.java]: check what on numbers? use meaningful name pls

Comment (Chezo) [reply Sum.java]: also use wrapper class in parameter

File: Sum.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        for (int i = 0; i < numbers.stream().count(); i++) {
```
Comment (Chezo) [inline Sum.java]: stream? if i accept the stream usage still wont be in for loop condition!

File: Sum.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            int currentNumber = numbers.get(i);
+            int numberCheckCount = 0;
+            for (int j = 0; j < numbers.stream().count(); j++) {
```
Comment (Chezo) [inline Sum.java]: stream? if i accept the stream usage still wont be in for loop condition!

Comment (Chezo) [reply Sum.java]: spoiler: numbers.size()?

File: Sum.java (hunk 1 above)
Comment (Chezo) [reply Sum.java]: i think there's should be return a list

**Review by Chezo — CHANGES_REQUESTED** [review]

File: Circle.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import java.util.Scanner;
+
+public class Circle {
+    static Scanner userInputScanner = new Scanner(System.in);
```
Comment (Chezo) [inline Circle.java]: static

File: Circle.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    public static double CirclePerimeter(int circleRadius){
```
Comment (Chezo) [inline Circle.java]: static

File: Circle.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        System.out.print("Enter the circle radius length:");
+        circleRadius = Integer.parseInt(userInputScanner.nextLine());
+        double circlePerimeter = 2*Math.PI*circleRadius;
+
+        return circlePerimeter;
+    }
+    public static double CircleArea(int circleRadius){
```
Comment (Chezo) [inline Circle.java]: static

File: ControlPanel.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import java.util.Scanner;
+
+public class ControlPanel {
+    static Scanner userInputScanner = new Scanner(System.in);
```
Comment (Chezo) [inline ControlPanel.java]: static

File: ControlPanel.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    static Circle mainCircle = new Circle();
```
Comment (Chezo) [inline ControlPanel.java]: static

File: ControlPanel.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    static Rectangle mainRectangle = new Rectangle();
```
Comment (Chezo) [inline ControlPanel.java]: static

File: ControlPanel.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    public static void controlPanelForOperations(int selectInputNumber){
```
Comment (Chezo) [inline ControlPanel.java]: static

File: Main.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
+import java.util.Scanner;
+public class Main {
+    static Scanner userInputScanner = new Scanner(System.in);
```
Comment (Chezo) [inline Main.java]: static

File: Main.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    static ControlPanel UserInputSelectOptionPanel = new ControlPanel();
```
Comment (Chezo) [inline Main.java]: static

File: Rectangle.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
+import java.util.Scanner;
+
+public class Rectangle {
+    static Scanner userInputScanner = new Scanner(System.in);
```
Comment (Chezo) [inline Rectangle.java]: static

File: Rectangle.java [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+    public static int RectanglePerimeter(int rectangleLateralFace, int anotherRectangleLateralFace){
```
Comment (Chezo) [inline Rectangle.java]: static

File: Rectangle.java [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+        System.out.print("Enter a number to calculate the rectangle perimeter:");
+        int SideOfaRectangle= Integer.parseInt(userInputScanner.nextLine());
+        System.out.print("Enter another number to calculate the rectangle perimeter:");
+        int AnotherSideOfaRectangle= Integer.parseInt(userInputScanner.nextLine());
+        int perimeterResult = (AnotherSideOfaRectangle+SideOfaRectangle)*2;
+
+        return perimeterResult;
+    }
+    public static int RectangleArea(int rectangleLateralFace, int anotherRectangleLateralFace){
```
Comment (Chezo) [inline Rectangle.java]: static

File: Rectangle.java [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+        System.out.print("Enter a number to calculate the rectangle area:");
+        int SideOfaRectangle = Integer.parseInt(userInputScanner.nextLine());
+        System.out.print("Enter another number to calculate the rectangle area:");
+        int AnotherSideOfaRectangle = Integer.parseInt(userInputScanner.nextLine());
+        int areaResult = SideOfaRectangle*AnotherSideOfaRectangle;
+
+        return areaResult;
+    }
+    public static double rectangleArcCoTangens(double RectangleArcCoTanegnt){
```
Comment (Chezo) [inline Rectangle.java]: static

File: src/pinCodeTasks/pinCodeTask1.java [hunk 17]
```diff
@@ -0,0 +1,37 @@
+package pinCodeTasks;
+
+import java.util.Scanner;
+
+public class pinCodeTask1 {
+
+    public static void main(String[] args) {
```
Comment (Chezo) [inline src/pinCodeTasks/pinCodeTask1.java]: static

**Review by Chezo — CHANGES_REQUESTED** [review]
Theres 3 main methods! MUST BE ONE! only Main class should have main method! Only main class should have maximum 2 static method, 1 of them the main!

File: Sum.java [hunk 18]
```diff
@@ -0,0 +1,38 @@
+import java.util.ArrayList;
+public class Sum {
+    public static void main(String[] args) {
+        int[] arrayNumbers = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
```
Comment (Chezo) [inline Sum.java:9]: should be private

File: Sum.java [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+        System.out.println(Sum.numbersSum(arrayNumbers));
+
+        ArrayList<Integer> numbers = new ArrayList<Integer>();
```
Comment (Chezo) [inline Sum.java]: Should be wrapper type List<Integer> numbers = new ArrayList<>();

File: Sum.java [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+        Sum.numbersCheck(numbers);
+    }
+    public static int numbersSum(int[] numbersArray){
+        int inputArrayNumbersSum = 0;
+        for(int i = 0; i < numbersArray.length; i++){
+            inputArrayNumbersSum += numbersArray[i];
+        }
+        return inputArrayNumbersSum;
+    }
+    public static void numbersCheck(ArrayList<Integer> numbersOfInputArray) {
```
Comment (Chezo) [inline Sum.java]: need better parameter name

File: Sum.java [hunk 21]
```diff
@@ -0,0 +1,38 @@
...
+        int[] inputNumbers = {1, 11, 22, 54, 21, 11, 78};
+        for(int i = 0; i < inputNumbers.length; i++){
+            numbers.add(inputNumbers[i]);
+        }
+        Sum.numbersCheck(numbers);
+    }
+    public static int numbersSum(int[] numbersArray){
+        int inputArrayNumbersSum = 0;
+        for(int i = 0; i < numbersArray.length; i++){
+            inputArrayNumbersSum += numbersArray[i];
```
Comment (Chezo) [inline Sum.java]: too specific in this scope

**Review by Chezo — CHANGES_REQUESTED** [review]

File: Sum.java (hunk 18 above)
Comment (Chezo) [reply Sum.java:9]: should be private

File: Sum.java [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+
+public class Sum {
+    public static void main(String[] args) {
+        Sum sum = new Sum();
+        int[] arrayNumbers = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+        System.out.println(sum.numbersSum(arrayNumbers));
+
+        List numbers = new ArrayList<>();
+        int[] inputNumbers = {1, 11, 22, 54, 21, 11, 78};
```
Comment (Chezo) [inline Sum.java]: should be private

**Review by Chezo — CHANGES_REQUESTED** [review]

**Review by Chezo — CHANGES_REQUESTED** [review]
nothing changed expect the resolved conversation without an ansver!

File: Sum.java (hunk 18 above)
Comment (Arens404) [reply Sum.java:9]: b2f08a76

File: Sum.java (hunk 22 above)
Comment (Arens404) [reply Sum.java]: b2f08a76

**Comment by Arens404** [comment]
> nothing changed expect the resolved conversation without an ansver!

b2f08a76

**Review by Chezo — APPROVED** [review]
acceptable

**Pull request merged**